File: weknot_android/app/src/main/java/com/example/weknot_android/view/fragment/OpenChatFragment.java
package com.example.weknot_android.view.fragment;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.weknot_android.R;
import com.example.weknot_android.base.BaseFragment;
import com.example.weknot_android.databinding.OpenChatFragmentBinding;
import com.example.weknot_android.viewmodel.OpenChatViewModel;
import com.example.weknot_android.widget.recyclerview.adapter.OpenChatAdapter;

public class OpenChatFragment extends BaseFragment<OpenChatFragmentBinding> {

    private OpenChatAdapter openChatAdapter;

    private OpenChatViewModel openChatViewModel;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        initViewModel();
        initData();

        observeOpenChatViewModel();
    }

    private void initViewModel() {
        openChatViewModel = ViewModelProviders.of(this).get(OpenChatViewModel.class);
    }

    private void initData() {
        openChatViewModel.getChattingRooms();
    }

    private void observeOpenChatViewModel() {
        openChatViewModel.getData().observe(this, openChatRooms -> {
            openChatAdapter = new OpenChatAdapter(getContext(), openChatRooms);
            setRecyclerView();
        });
    }

    private void setRecyclerView() {
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        binding.chatRoomRecyclerview.setAdapter(openChatAdapter);
        binding.chatRoomRecyclerview.setLayoutManager(linearLayoutManager);
    }

    @Override
    protected int layoutId() {
        return R.layout.open_chat_fragment;
    }
}

File: weknot_android/app/src/main/java/com/example/weknot_android/view/fragment/OpenChatFragment.kt
package com.example.weknot_android.view.fragment

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.lifecycle.Observer
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.example.weknot_android.BR
import com.example.weknot_android.R
import com.example.weknot_android.base.BaseDialog
import com.example.weknot_android.base.fragment.BaseListFragment
import com.example.weknot_android.databinding.OpenChatFragmentBinding
import com.example.weknot_android.view.activity.ChatActivity
import com.example.weknot_android.view.activity.ProfileActivity
import com.example.weknot_android.view.dialog.CreateRoomDialog
import com.example.weknot_android.viewmodel.OpenChatViewModel
import com.google.android.material.floatingactionbutton.FloatingActionButton

class OpenChatFragment : BaseListFragment<OpenChatFragmentBinding, OpenChatViewModel>() , SwipeRefreshLayout.OnRefreshListener  {

    override fun getLayoutId(): Int {
        return R.layout.open_chat_fragment
    }

    override fun getViewModel(): Class<OpenChatViewModel> {
        return OpenChatViewModel::class.java
    }

    override fun getBindingVariable(): Int {
        return BR.viewModel
    }

    override fun initObserver() {
        with(viewModel) {
            onErrorEvent.observe(this@OpenChatFragment, Observer {
                simpleToast(it.message)
            })

            openCreateRoom.observe(this@OpenChatFragment, Observer {
                newInstance(createRoomDialog).show(fragmentManager)
            })

            createRoomDialog.dialogCloseEvent.observe(this@OpenChatFragment, Observer {
                getChattingRooms()
            })

            openChatAdapter.openChatRoom.observe(this@OpenChatFragment, Observer {
                val intent = Intent(context, ChatActivity::class.java)
                intent.putExtra("key", it)
                startActivityWithFinish(intent)
            })
        }
    }

    override fun setBtn(): FloatingActionButton {
        return binding.createBtn
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setUp()
    }

    override fun onResume() {
        super.onResume()
        viewModel.getChattingRooms()
    }

    override fun onRefresh() {
        viewModel.getChattingRooms()
        binding.swipeRefreshLayout.isRefreshing = false
    }

    private fun setUp() {
        binding.chatRoomRecyclerview.addOnScrollListener(scrollListener)

        binding.swipeRefreshLayout.setColorSchemeResources(R.color.colorPrimary, R.color.colorAccent)
        binding.swipeRefreshLayout.setOnRefreshListener(this)
    }
}